### PR TITLE
[libmicrohttpd] Move to autoconfigure instead of cmake

### LINKED ISSUE #6516: [libmicrohttpd] Failed to install on Ubuntu 16.04
```
$ ./vcpkg install libmicrohttpd

The following packages will be built and installed:
    libmicrohttpd[core]:x64-linux
Starting package 1/1: libmicrohttpd:x64-linux
Building package libmicrohttpd[core]:x64-linux... 
-- Using cached /home/vcpkg/vcpkg/downloads/libmicrohttpd-0.9.63.tar.gz
-- Configuring x64-linux-dbg
-- Configuring x64-linux-rel
-- Building x64-linux-dbg
CMake Error at scripts/cmake/vcpkg_build_cmake.cmake:202 (message):
    Command failed: /home/vcpkg/vcpkg/downloads/tools/cmake-3.14.0-linux/cmake-3.14.0-Linux-x86_64/bin/cmake;--build;.;--config;Debug;--target;install;--;-v
    Working Directory: /home/vcpkg/vcpkg/buildtrees/libmicrohttpd/x64-linux-dbg
    See logs for more information:
      /home/vcpkg/vcpkg/buildtrees/libmicrohttpd/install-x64-linux-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_install_cmake.cmake:24 (vcpkg_build_cmake)
  ports/libmicrohttpd/portfile.cmake:21 (vcpkg_install_cmake)
  scripts/ports.cmake:71 (include)


Error: Building package libmicrohttpd:x64-linux failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: libmicrohttpd:x64-linux
  Vcpkg version: 2018.11.23-unknownhash

Additionally, attach any relevant sections from the log files above.
```

Log file: [install-x64-linux-dbg-out.log](https://github.com/microsoft/vcpkg/files/3193808/install-x64-linux-dbg-out.log)

### PR DESCRIPTION
**Pull request by NoFr1ends**
**Describe the pull request**
The libmicrohttpd autoconfig is pretty complex and the current reimplementation of it only has support for MSVC and is missing features. This PR moves back to use the official build system of libmicrohttpd

- What does your PR fix?
Fixes #6516

### CODE REVIEW

File: ports/libmicrohttpd/build_windows.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+fi
+
+PATH_TO_BUILD_DIR=$1
+PATH_TO_SRC_DIR=$2
+PATH_TO_PACKAGE_DIR=$3
+# Note: $4 is extra configure options
+
+cd "$PATH_TO_BUILD_DIR"
+echo "=== CONFIGURING ==="
+"$PATH_TO_SRC_DIR/configure" --toolchain=msvc "--prefix=$PATH_TO_PACKAGE_DIR" $4
```
Comment (Neumann-A) [inline ports/libmicrohttpd/build_windows.sh:19]: `prefix` can only point to `PATH_TO_PACKAGE_DIR` if the library is not hardcoding some stuff with that path. Else you have to set prefix to `INSTALLED_DIR `and use `DESTDIR` in `make install`

File: ports/libmicrohttpd/build_windows.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+echo "=== BUILDING ==="
+make -j6
```
Comment (Neumann-A) [inline ports/libmicrohttpd/build_windows.sh:21]: `-j6` instead make use of `VCPKG_CONCURRENCY`

**Comment by LilyWangL** [comment]
Thanks for your PR. Could you please remove the following codes in _scripts/ci.baseline.txt_?
```
libmicrohttpd:x64-linux=fail
libmicrohttpd:x64-osx=fail
```

**Comment by LilyWangL** [comment]
Pinging @NoFr1ends for response. Is work still being done for this PR?

**Comment by LilyWangL** [comment]
Closing this PR since it seems that no progress is being made. Please reopen if work is still being done.

**Pull request closed without merging**